osf/management/commands/force_archive.py

```diff
@@ -280,48 +281,72 @@ def get_logs_to_revert(reg):
             Q(node=reg.registered_from) |
             (Q(params__source__nid=reg.registered_from._id) | Q(params__destination__nid=reg.registered_from._id))).order_by('-date')
 
+def get_file_obj_from_log(log, reg):
+    try:
+        return BaseFileNode.objects.get(_id=log.params['urls']['view'].split('/')[4])
+    except KeyError:
+        if log.action == 'osf_storage_folder_created':
+            return OsfStorageFolder.objects.get(
+                target_object_id=reg.registered_from.id,
+                name=log.params['path'].split('/')[-2]
+            )
+        elif log.action == 'osf_storage_file_removed':
+            path = log.params['path'].split('/')
+            return TrashedFileNode.objects.get(
+                target_object_id=reg.registered_from.id,
+                name=path[-1] or path[-2]  # file name or folder name
+            )
```

Comment:
This also doesn't guarantee uniqueness for the `.get`, and is much more likely to raise `MultipleObjectsReturned`. You can create and delete a file with `name='test.txt'` 10 times and generate 10 `TrashedFile` objects which would all be returned by this. 

You want to get the `TrashedFileNode` matching this query which was `.deleted` just prior to this `log.created`